gobblin-modules/gobblin-compliance/src/main/java/gobblin/compliance/ComplianceRecord.java

```diff
@@ -0,0 +1,258 @@
+    if (!st.endsWith(SINGLE_QUOTE)) {
+      quotedString += SINGLE_QUOTE;
+    }
+    return quotedString;
+  }
+
+  /**
+   * This method splits the multiple partitions separated by '/' to list.
+   * datepartition=2016-01-01-00/size=12345 will be [datepartition=2016-01-01-00, size=12345]
+   * All "/" in the partition values are converted to %2 by Hive.
+   */
+  public List<String> getPartitionValues() {
+    return ImmutableList.<String>builder().addAll(SLASH_SPLITTER.splitToList(getPartitionName())).build();
+  }
+
+  /**
+   * This method splits the string on the basis of '=' into key value pairs and also add quotes to the values.
+   * [datepartition=2016-01-01-00, size=12345] will be [datepartition : '2016-01-01-00', size : '12345']
+   * All "=" in the partition values are converted to %3 by Hive.
+   */
+  public Map<String, String> getPartitionMap() {
+    Map<String, String> partitionMap = new LinkedHashMap<>();
+    for (String st : getPartitionValues()) {
+      final List<String> list = EQUALITY_SPLITTER.splitToList(st);
+      Preconditions.checkArgument(list.size() == 2, "Wrong partitioned value " + st);
+      String key = list.get(0);
+      String value = list.get(1);
+      value = getQuotedString(value);
+      partitionMap.put(key, value);
+    }
+    return partitionMap;
+  }
+
+  /**
+   * This method returns partitioning column names separated by comma.
+   * [datepartition=2016-01-01-00, size=12345] will be datepartition, size
+   */
+  public String getCommaSeparatedPartitionColumnNames() {
+    StringBuilder sb = new StringBuilder();
+    for (String partitionName : getPartitionMap().keySet()) {
+      if (!sb.toString().isEmpty()) {
+        sb.append(",");
+      }
+      sb.append(partitionName);
+    }
+    return sb.toString();
+  }
+
+  /**
+   * This method builds the where clause for the insertion query.
+   * If prefix is a, then it builds a.datepartition='2016-01-01-00' AND a.size='12345' from [datepartition : '2016-01-01-00', size : '12345']
+   */
+  public String getWhereClauseForPartition(String prefix) {
+    StringBuilder sb = new StringBuilder();
+    for (String partitionName : getPartitionMap().keySet()) {
+      if (!sb.toString().isEmpty()) {
+        sb.append(" AND ");
+      }
+      sb.append(prefix + partitionName);
+      sb.append("=");
+      sb.append(getPartitionMap().get(partitionName));
+    }
+    return sb.toString();
+  }
+
+  /**
+   * This method returns the partition spec of the partition.
+   * Example : (datepartition='2016-01-01-00', size='12345')
+   */
+  public String getPartitionSpec() {
+    StringBuilder sb = new StringBuilder();
+    for (String partitionName : getPartitionMap().keySet()) {
+      if (!sb.toString().isEmpty()) {
+        sb.append(",");
+      }
+      sb.append(partitionName);
+      sb.append("=");
+      sb.append(getPartitionMap().get(partitionName));
+    }
+    return sb.toString();
+  }
+
+  public static class ComplianceRecordBuilder {
```

Comment:
nit: Have you checked `@builder`? https://projectlombok.org/features/Builder.html